### PR TITLE
Added support for topic name list

### LINKED ISSUE #113: How can I list topics?
Hello, good work with this library. My question is self-described, I want to list all the available topics and their partitions, or all partitions -> topics, like the `Client.Topics()` of sarama package. Thank you.

### PR DESCRIPTION
**Pull request by lruggieri**
Enable users to get a list of topics by doing

```
reader := NewReader(config)
topics, err := reader.ReadTopicNames(context.Background())
```

Resolve #113

### CODE REVIEW

**Comment by snowbeta** [comment]
Is there anything that needs to be done for this to be merged?

**Comment by alok87** [comment]
please merge and release this

File: reader.go [hunk 1]
```diff
@@ -1056,6 +1056,26 @@ func (r *Reader) readLag(ctx context.Context) {
...
+	if len(r.config.Brokers) == 0{
+		return nil, errors.New("no broker provided")
+	}
+
+	conn, err := r.config.Dialer.DialLeader(ctx, "tcp", r.config.Brokers[0], r.config.Topic, r.config.Partition)
+	if err != nil {
+		return nil, err
+	}
+	defer conn.Close()
+	topicNames, err := conn.Topics()
```
Comment (deliangyang) [inline reader.go:1069]: Why not do it directly?

return conn.Topics()

Comment (deliangyang) [inline reader.go:1069]: Why not do it directly?
```go
return conn.Topics()
```

Comment (lruggieri) [reply reader.go]: good point, thanks

**Comment by achille-roussel** [comment]
Hello @deliangyang, thanks for the contribution.

We've added a section in the README a little while back to demonstrate how to get the topic list from the metadata requests (which returns a list of partitions) https://github.com/segmentio/kafka-go#to-list-topics

Would this be sufficient to help developers in your opinion? Or do you think there's value in have a special API for this?

**Comment by achille-roussel** [comment]
This should have been solved by https://github.com/segmentio/kafka-go/pull/663, feel free to reopen if that is not the case 👍

**Pull request closed without merging**